app/src/main/java/com/markkolenbrander/capstonenewsapp/MainActivity.kt

```diff
@@ -0,0 +1,136 @@
+package com.markkolenbrander.capstonenewsapp
+
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.widget.TextView
+import androidx.core.view.children
+import com.markkolenbrander.capstonenewsapp.databinding.ActivityMainBinding
+
+class MainActivity : AppCompatActivity() {
+
+    private lateinit var binding: ActivityMainBinding
+    private val articlesCollection = mutableMapOf<String, MutableList<String?>>()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivityMainBinding.inflate(layoutInflater)
+        val view = binding.root
+        setContentView(view)
+
+        addArticles()
+        makingTextViews(articlesCollection)
+    }
+
+    private fun addArticles(){
```

Comment:
No change required here, but you could rename and refactor this method to map directly, rather than modify `articlesCollection`. So, something like this:

```suggestion
    private fun createArticlesMap(): Map<String, MutableList<String?>> {
```

This way, `articlesCollection` as a variable wouldn't be required.

Reply:
Yes good one, next time I will do it like that